enterprise/internal/authz/azuredevops/provider.go

```diff
@@ -0,0 +1,173 @@
+package azuredevops
+
+import (
+	"context"
+	"time"
+
+	authztypes "github.com/sourcegraph/sourcegraph/enterprise/internal/authz/types"
+	"github.com/sourcegraph/sourcegraph/enterprise/internal/licensing"
+	"github.com/sourcegraph/sourcegraph/internal/authz"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/auth"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/azuredevops"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type Provider struct {
+	db database.DB
+
+	urn      string
+	codeHost *extsvc.CodeHost
+
+	orgs     []string
+	projects []string
+}
+
+func (p *Provider) FetchAccount(ctx context.Context, user *types.User, _ []*extsvc.Account, _ []string) (*extsvc.Account, error) {
+	return nil, nil
+}
+
+func (p *Provider) FetchUserPerms(ctx context.Context, account *extsvc.Account, opts authz.FetchPermsOptions) (*authz.ExternalUserPermissions, error) {
+	if account == nil {
+		// TODO: User auth form code host config
+	}
+
+	_, token, err := azuredevops.GetExternalAccountData(ctx, &account.AccountData)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"failed to get external account data external account with ID: %d, this might be related to a bad JSON in the database table user_external_accounts for this ID or a misconfigured encryption key",
```

Comment:
Is this a database error? Because the first sentence makes it sound like we're talking to an external service. 

As for second sentence: it could literally be anything if it's a database error - connection error, timeout, computer dies, ... I'd simplify to 
```suggestion
			"failed load external account data from database for external account with ID: %d",
```